extras/menus/advancedvsh/advanced.c

```diff
@@ -46,385 +46,416 @@ extern char* available_fonts[];
+			// Light Purple 
+			case 17:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0993366;
+				blit_set_color(fc,bc);
+				break;
+			// Teal 
+			case 18:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00808000;
+				blit_set_color(fc,bc);
+				break;
+			// Light Teal 
+			case 19:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0808000;
+				blit_set_color(fc,bc);
+				break;
+			// Aqua 
+			case 20:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00cccc00;
+				blit_set_color(fc,bc);
+				break;
+			// Light Aqua 
+			case 21:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0cccc00;
+				blit_set_color(fc,bc);
+				break;
+			// Grey 
+			case 22:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00737373;
+				blit_set_color(fc,bc);
+				break;
+			// Light Grey 
+			case 23:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0737373;
+				blit_set_color(fc,bc);
+				break;
+			// Black 
+			case 24:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00000000;
+				blit_set_color(fc,bc);
+				break;
+			// Light Black 
+			case 25:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0000000;
+				blit_set_color(fc,bc);
+				break;
+			// Light Red
+			case 26: 
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa00000ff;
+				blit_set_color(fc,bc);
+				break;
+			// Red
+			case 27:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0x000000ff;
+				blit_set_color(fc,bc);
+				break;
+			// Light White  
+			case 28:
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0xafffffff;
+				blit_set_color(fc,bc);
+				break;
+			default:	
+				fc = (submax_menu==submenu_sel) ? 0xffffff : 0x0000a5ff;
+				blit_set_color(fc,bc);
+		}
 
+		// display menu
+		if(g_messages[MSG_USB_DEVICE + submax_menu]) {		
+			int submenu_start_x, submenu_start_y;
 			subcur_menu = submax_menu;
-			blit_string(xPointer, (pointer[5] + subcur_menu)*8, msg);
-			msg = subitem_str[submax_menu];
-
-			if (submax_menu == SUBMENU_UMD_REGION_MODE) {
-				if(psp_model == PSP_GO || IS_VITA_ADR(ark_config)) msg = g_messages[MSG_UNSUPPORTED];
-				blit_string( (pointer[6] * 8) + 128, (pointer[5] + subcur_menu)*8, msg);
-			}
-			else if(msg) {
-				blit_string( (pointer[6] * 8) + 128, (pointer[5] + subcur_menu)*8, msg);
+			
+			// set the y position
+			submenu_start_y = (pointer[5] + subcur_menu) * 8;
+			
+			int width = 0, padding = 0, temp = 0, i;
+			// submenus between USB_DEVICE and UMD_REGION are the only ones with subitems
+			if (submax_menu >= SUBMENU_USB_DEVICE && submax_menu <= SUBMENU_UMD_REGION_MODE) {
+				// check if there's a subitem and print it
+				int subitem_start_x = (pointer[6] * 8) + 128;
+				if(subitem_str[submax_menu]) {
+					// check if PSP Go or PSVita because UMD Region mode is unsupported on them
+					if ((psp_model == PSP_GO || IS_VITA_ADR(ark_config)) && submax_menu == SUBMENU_UMD_REGION_MODE){
+						// write the unsupported string
+						scePaf_snprintf(msg, 128, "%s", g_messages[MSG_UNSUPPORTED]);
```

Comment:
UNSUPPORTED is only 11 bytes, 128 is overkill.

Reply:
I just went with consistency. All other snprintf are set to a max size of 128 too, like the size of the `msg` buffer.